msal/token_cache.py

```diff
@@ -243,7 +243,7 @@ def __add(self, event, now=None):
                     at["refresh_on"] = str(now + refresh_in)  # Schema wants a string
                 self.modify(self.CredentialType.ACCESS_TOKEN, at, at)
 
-            if client_info and not event.get("skip_account_creation"):
+            if (client_info or iwa) and not event.get("skip_account_creation"):
```

Comment:
The access token returned by IWA did not meet the criteria to be added as an `ACCOUNT` credential type in the `TokenCache` i.e. it didn't have `client_info` in the response. It was being added as a `ACCESS_TOKEN` credential type. The cache test in the e2e tests were failing because of it since the `get_accounts()` call didn't return anything.  Hence, I added a special case of `iwa` to be added to the `ACCOUNT` credential type.